src/TraceEvent/TraceLoggingEventId.cs

```diff
@@ -10,19 +11,20 @@ namespace Microsoft.Diagnostics.Tracing
     /// <summary>
     /// TraceLoggingEvnetId is a class that manages assigning event IDs (small 64k numbers)
     /// to TraceLogging Style events (which don't have them).  Because TraceEvent uses EventIDs
-    /// so fundamentally this difficiency is very problematic.   
+    /// so fundamentally this deficiency is very problematic.   
     /// 
     /// Arguably this should have been done by the ETW system itself.  
     /// 
     /// You use it by calling TestForTraceLoggingEventAndFixupIfNeeded on eventRecords.  
     /// You also have to explicitly call 'Dispose' when you are done with this class.  
     /// </summary>
-    internal unsafe struct TraceLoggingEventId
+    internal unsafe struct TraceLoggingEventId : IDisposable
     {
         /// <summary>
         /// Checks to see if eventRecord has TraceLogging meta data associated with it (EVENT_HEADER_EXT_TYPE_EVENT_SCHEMA_TL)
         /// and if so updates EventHeader.Id to be an event ID unique to that provider/opcode/meta-data blob. 
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I didn't try to check if the method is inlined or not in runtime. Since I see @vancem has changed TraceEvent lib to target 4.5 and therefore I made the change.
For 3.5 support, do we consider to create a release branch and only do bug fix as needed?